unit_testing/test_sct_compute_mtsat.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+
+from __future__ import print_function, absolute_import
+
+import os
+import sys
+
+from spinalcordtoolbox.utils import sct_test_path
+from spinalcordtoolbox import __sct_dir__
+sys.path.append(os.path.join(__sct_dir__, 'scripts'))
+
+import sct_compute_mtsat
+
+
+class InputParam1:
+    mt = sct_test_path('mt', 'mt1.nii.gz')
+    pd = sct_test_path('mt', 'mt0.nii.gz')
+    t1 = sct_test_path('mt', 't1w.nii.gz')
+    trmt = None
+    trpd = None
+    trt1 = None
+    famt = None
+    fapd = None
+    fat1 = None
+
+
+class InputParam2:
+    mt = sct_test_path('mt', 'mt1.nii.gz')
+    pd = sct_test_path('mt', 'mt0.nii.gz')
+    t1 = sct_test_path('mt', 't1w.nii.gz')
+    trmt = 1
+    trpd = 2
+    trt1 = 3
+    famt = 4
+    fapd = 5
+    fat1 = 6
+
+
+def test_get_tr_and_flipangle():
+    assert sct_compute_mtsat.get_tr_and_flipangle(InputParam1()) == (0.03, 0.03, 0.015, 9, 9, 15)
+    assert sct_compute_mtsat.get_tr_and_flipangle(InputParam2()) == (1, 2, 3, 4, 5, 6)
+
+
+def test_with_json_sidecar():
+    sct_compute_mtsat.main(['-mt', sct_test_path('mt', 'mt1.nii.gz'),
+                            '-pd', sct_test_path('mt', 'mt0.nii.gz'),
+                            '-t1', sct_test_path('mt', 't1w.nii.gz')])
+    # Check if output file exists
+    assert os.path.isfile(sct_test_path('mt', 'mtsat.nii.gz'))
```

Comment:
Would this be better as an integrity test in `testing/`, considering that's where the other SCT script tests are located?

Reply:
indeed, but i think we decided (I can't find the discussion thread-- maybe it was in SCT meeting?) to slowly get rid of the `testing/` folder, and instead rely on the `unit_testing` for the long term, as it is more aligned with Python conventions. Integrity testing could also be included in unit_testing, with prefix test_sct_*** (instead of test_module***).

Reply:
Ahh, thank you for the clarification!

> i think we decided (I can't find the discussion thread-- maybe it was in SCT meeting?) 

In the future, when a decision like this is made during a meeting, perhaps it could be an action item for someone to update the GitHub Wiki? @Drulex has already taken some great initiative in doing something similar (https://github.com/neuropoly/spinalcordtoolbox/pull/2875#discussion_r483724457) and it would be lovely if we could make this a habit. There's a chance there are other nuggets of information regarding SCT's design that are hidden away in issues and meetings, but that could also be on the Wiki. :stuck_out_tongue: